### DISCUSSION TITLE
Nest a variable under a specific level/value of another variable

### DISCUSSION THREAD

**Discussion by tt05-hod** [insightsengineering/rtables#879] [Q&A] [answered]
Hi team, apologies if this is somewhere in the loads of helpful documentation and I overlooked the answer. Assume that I have a survey question that asks about gender identity. One of the options is "Prefer to self-describe" where then the user enters in text their description. When the data are collected, this open-ended response is stored as a separate variable. Is it possible to nest the levels (assume in this case there are only a few self-descriptions) of the open-ended response underneath the "Prefer to self-describe" level of the original question? The motivation here is to replicate visualizations previously completed in SAS at the request of a stakeholder.

Please see the following dummy data and result/target outputs:

Data:
```
survey_data <- structure(list(Gender_Identity = structure(c(4L, 1L, 3L, 3L, 
3L, 2L, 1L, 2L, 4L, 3L, 4L, 1L, 4L, 3L, 2L, 1L, 1L, 2L, 4L, 1L, 
2L, 2L, 1L, 3L, 3L), levels = c("Man", "Non-binary", "Prefer to self-describe", 
"Woman"), class = "factor"), Self_Describe = structure(c(NA, 
NA, 1L, 2L, 1L, NA, NA, NA, NA, 1L, NA, NA, NA, 1L, NA, NA, NA, 
NA, NA, NA, NA, NA, NA, 3L, 1L), levels = c("Agender", "Genderqueer", 
"Other"), class = "factor"), Age_Group = structure(c(1L, 5L, 
5L, 3L, 2L, 5L, 6L, 5L, 3L, 3L, 1L, 4L, 2L, 1L, 2L, 4L, 5L, 1L, 
1L, 1L, 5L, 6L, 6L, 6L, 3L), levels = c("18-24", "25-34", "35-44", 
"45-54", "55-64", "65+"), class = "factor"), Race = structure(c(1L, 
2L, 2L, 1L, 2L, 1L, 1L, 2L, 4L, 1L, 4L, 3L, 1L, 2L, 1L, 3L, 2L, 
2L, 2L, 4L, 1L, 1L, 3L, 1L, 1L), levels = c("Asian", "Black or African American", 
"Other", "White"), class = "factor")), class = c("tbl_df", "tbl", 
"data.frame"), row.names = c(NA, -25L))

library(rtables)

basic_table() |> 
  analyze(vars = c("Gender_Identity", "Self_Describe", "Age_Group", "Race")) |> 
  build_table(survey_data)

```
Result:
```
                              all obs
—————————————————————————————————————
Gender_Identity                      
  Man                            7   
  Non-binary                     6   
  Prefer to self-describe        7   
  Woman                          5   
Self_Describe                        
  Agender                        5   
  Genderqueer                    1   
  Other                          1   
Age_Group                            
  18-24                          6   
  25-34                          3   
  35-44                          4   
  45-54                          2   
  55-64                          6   
  65+                            4   
Race                                 
  Asian                         11   
  Black or African American      8   
  Other                          3   
  White                          3 
```

Target output:
```
                              all obs
—————————————————————————————————————
Gender_Identity                      
  Man                            7   
  Non-binary                     6   
  Prefer to self-describe        7 
    Agender                      5   
    Genderqueer                  1   
    Other                        1   
  Woman                          5   
Age_Group                            
  18-24                          6   
  25-34                          3   
  35-44                          4   
  45-54                          2   
  55-64                          6   
  65+                            4   
Race                                 
  Asian                         11   
  Black or African American      8   
  Other                          3   
  White                          3 
```

Note: This is not a real survey question. The example was contrived by ChatGPT to help illustrate my actual problem.

**Comment by Melkiades** [comment, marked as answer]
this is what you are looking for:
```r
basic_table() |>
  split_rows_by("Gender_Identity", label_pos = "visible") |>
  summarize_row_groups(format = "xx") |>
  analyze(vars = c("Self_Describe")) |>
  analyze(vars = c("Age_Group", "Race"), nested = FALSE) |>
  build_table(survey_data) %>% 
  trim_rows()
```

**Reply by tt05-hod** [reply]
Thank you for this! My actual problem is more complex, so I can take you answer and try to build upon it. A couple of differences I need to account for:
- The gender_identity/self_describe items are in between other variables being analyzed (i.e. not the first nor last items in the table)
- trim_rows() removes other data I need to keep. I see `prune_table` has some additional functions to explore here

**Comment by gmbecker** [comment]
This is one of the few things that legitimately runs afoul of the constraints on the layouting engine (particularly once the PR for higher level Ns is merged in, as that was one of the other longstanding limitations).

That said, you have a few options for how to get a table to render like that. The first is to write an afun that emits allof the following rows and assigns the appropriate indent modifiers to them:

```
Gender_Identity                      
  Man                            7   
  Non-binary                     6   
  Prefer to self-describe        7 
    Agender                      5   
    Genderqueer                  1   
    Other                        1   
  Woman                          5   
```

As in that whole block will be a single analyze call that just looks like a split with partial analyze under it. This will work, though the row paths will be somewhat unintuitive and wanting to sort the rows under "prefer to self-describe" would be difficult at best.

The other option is to construct 3 parts and just hid ethat fact from the reader of the table:

something along the lines (untested)
```
lyt <- basic_table() %>%
   analye("SEX") %>% ## only emit rows before prefer not to state
   split_rows_by("SEX", split_fun = keep_split_values("PREFERNOTTOSTATE")) %>%
   analyze("SELFDESCRIBE") %>% ## 
   analyze("SEX", nested = FALSE) %>% ## whatever rows still need to be emited
   <rest of table...>
```

The table structure wil be weird here and again, dynamic sorting would be troublesome, but it will look like you want it to when you get the label visibliity and indent mod controls right.

If I had to make that table tomorrow I'd probably go with option one, but it depends some on what the downstream processing is.
